packages/common/src/featureServiceHelpers.ts

```diff
@@ -1212,6 +1223,58 @@ export function _getFieldNames(
   }
 }
 
+/**
+ * Get a list of any dynamically calculated fields
+ * These fields are still valid but will not exist in the source service
+ *
+ * @param table the table instance to compare
+ *
+ * @returns an array of field names
+ * @private
+ */
+export function _getDynamicFieldNames(
+  table: any
+): string[] {
+  const sourceFieldInfos = {};
+  const fields = [];
+  // get all the expressions as we will need to check if any of the fields are referenced
+  // also store any fields that share a source and have an expression
+  const expressions = table.sourceLayerFields.reduce((prev,cur) => {
+    /* istanbul ignore else */
+    if (cur.source) {
+      const hasExpression = cur.expression ? true : false;
+      if (Object.keys(sourceFieldInfos).indexOf(cur.source) > -1) {
+        const source = sourceFieldInfos[cur.source];
+        source.hasExpression = source.hasExpression || hasExpression;
+        source.names.push(cur.name);
+        /* istanbul ignore else */
+        if (source.hasExpression || hasExpression) {
```

Comment:
This line seems unnecessary because `source.hasExpression` has already been adjusted on line 1248